plugins/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -142,6 +190,39 @@
 from ansible_collections.community.general.plugins.module_utils.gitlab import auth_argument_spec, gitlab_authentication
 
 
+def vars_to_variables(vars):
+    # transform old vars to new variables structure
+    variables = list()
+    for item in vars.keys():
+        if (isinstance(vars.get(item), str) or
+           isinstance(vars.get(item), int) or
+           isinstance(vars.get(item), float)):
+            variables.append(
+                {
+                    "name": item,
+                    "value": str(vars.get(item))
+                }
+            )
+        else:
```

Comment:
The old code checked whether it was a dictionary, or called `module.fail_json` otherwise. The new code should do the same.